src/variant_dictionary.rs

```diff
@@ -141,6 +141,9 @@ impl VariantDictionary {
             data.extend_from_slice(&field_buffer);
         }
 
+        // signify end of variant dictionary
+        data.push(0);
```

Comment:
@sseemayer does that mean that we were also not verifying that when parsing? Should we add it?

Reply:
Yes, we should. There will be probably more details like this that will be uncovered before files generated by us can be opened by other KeePass implementations.

Reply:
resolved with da4c9d5 .